### ISSUE TITLE
Ping fails often and "system error" message is displayed

### ISSUE THREAD

**Issue by quarcko** [resoai/TileBoard#310]
Hi,

Firstly want to thank you for a great dashboard, that i'm currently trying to set up.
But i have a slight annoyance with ping.
(i have tried both over the internet and on local network only, it seems that it does not matter).

So the ping fails like once every minute. and i get 3 messages in a row:
1) Ping failed (orange)
2) Reconnect successful (green)
3) System error (red)

always in this sequence.

So now i got into sources and i'm trying different settings:
ping every 60 seconds, with a timeout of 10 seconds.

does anybody else experience constant ping failures?
HA is latest 0.102.X

**Comment by rchl** [comment]
If it happens on your desktop, can you check with (Chrome) inspector open? You could look into `Network` tab and check exactly what kind of network failure you are getting.

**Comment by quarcko** [comment]
Yes, i use it on Desktop, so in the console i see following info:

api.js:195 WebSocket connection to 'wss://xxxxxxx/api/websocket' failed: Data frame received after close

dataObject
data: Event
bubbles: false
cancelBubble: false
cancelable: false
composed: false
currentTarget: WebSocket {url: "wss://xxxxxxx/api/websocket", readyState: 3, bufferedAmount: 0, onopen: null, onerror: null, …}
defaultPrevented: false
eventPhase: 0
isTrusted: true
path: []
returnValue: true
srcElement: WebSocket {url: "wss://xxxxxxx/api/websocket", readyState: 3, bufferedAmount: 0, onopen: null, onerror: null, …}
target: WebSocket {url: "wss://xxxxxxx/api/websocket", readyState: 3, bufferedAmount: 0, onopen: null, onerror: null, …}
timeStamp: 98481.56499999459
type: "error"
__proto__: Event
message: "System error"

**Comment by quarcko** [comment]
I have added some console.log into "main.js":

`function pingConnection () {
      if(!$scope.ready || realReadyState === false) return; // no reason to ping if unready was fired

      var timeout = 3000;

      var success = false;

      Api.sendPing(function (res) {
	 console.log("PONG received");
         if('id' in res) success = true;
      });

      console.log("PING sent");

      setTimeout(function () {
	 console.log("Timeout! is PONG?");
         if(success) return;

         realReadyState = false;`

**Comment by quarcko** [comment]
So basically it goes like this in console (during failure):
1) PING sent
2) Timeout! is PONG?
3) api.js:195 WebSocket connection to 'wss://home.gcc.lt/api/websocket' failed: Data frame received after close

sometimes there is no "PONG" response, and reconnection gives this error.

**Comment by rchl** [comment]
Connection to the API is interrupted, most likely. Hard to say why. Maybe you have some proxy in between that kills long-running connections.

As I said, maybe looking at `Network` tab and failing websocket connection would give some clue.

**Comment by quarcko** [comment]
Very strange, but it happens in very different configurations exactly the same:

1) From work computer (HA is via NGINX ovet https and wss)
2) From home computer direct HTTP port 8123 and ws
3) From ubuntu tablet (also direct HTTP port 8123 and ws)

In all configurations it is consistent - once a minute ping fails, reconnects and error....


so it seems like a problem in HA itself.. what else?

**Comment by samtwilliams** [comment]
@quarcko did you ever get this resolved? I have exactly the same issue, had it on a RpI and also Ubuntu container setup.

I also ruled out WiFi issues by connecting direct to my switch network.

**Comment by quarcko** [comment]
Hi,

no i never found the real cause, but i just disabled ping function in config and my panel works super good without it.
it never gets disconnected.
so apparently there is a bug in this ping routine,
but as practice shows it is not needed at all.

Q.

⁣Get BlueMail for Android ​
